crates/hir/src/lib.rs

```diff
@@ -2192,45 +2220,175 @@ fn expr_store_diagnostics<'db>(
         .macro_calls()
         .for_each(|(_ast_id, call_id)| macro_call_diagnostics(db, call_id, acc));
 }
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+enum AnyFunctionId {
+    FunctionId(FunctionId),
+    BuiltinDeriveImplMethod { method: BuiltinDeriveImplMethod, impl_: BuiltinDeriveImplId },
+}
+
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
```

Comment:
Let's "inline" the debug impl here and delegate it to `AnyFunctionId` to get rid of the double FunctionId printing (since we have test fixtures that print those and its just noisy)